### PR TITLE
Update TiDB specific function docs

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

1. Make sure the page TOC links to the functions 
2. Remove argument description from the page TOC
3. Add info for `VITESS_HASH()`
4. Reorganize the functions in alphabetical order

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/17693
- Other reference link(s): https://github.com/pingcap/tidb/issues/20971

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/tidb-functions.md [hunk 1]
```diff
@@ -9,34 +9,147 @@ summary: 学习使用 TiDB 特有的函数。
...
+| [`TIDB_DECODE_PLAN()`](#tidb_decode_plan) | 用于解码 TiDB 执行计划。 |
+| [`TIDB_DECODE_SQL_DIGESTS()`](#tidb_decode_sql_digests) | 用于在集群中查询一组 SQL digest 所对应的 SQL 语句的归一化形式（即去除格式和参数后的形式）。 |
+| [`TIDB_ENCODE_SQL_DIGEST()`](#tidb_encode_sql_digest) | 用于为查询字符串获取 digest。 |
+| [`TIDB_IS_DDL_OWNER()`](#tidb_is_ddl_owner) | 用于检查你连接的 TiDB 实例是否是 DDL Owner。DDL Owner 是代表集群中所有其他节点执行 DDL 语句的 TiDB 实例。 |
+| [`TIDB_PARSE_TSO()`](#tidb_parse_tso) | 用于从 TiDB TSO 时间戳中提取物理时间戳。参见 [`tidb_current_ts`](/system-variables.md#tidb_current_ts)。 |
+| [`TIDB_PARSE_TSO_LOGICAL()`](#tidb_parse_tso_logical) | 用于从 TiDB TSO 时间戳中提取逻辑时间戳。|
+| [`TIDB_ROW_CHECKSUM()`](#tidb_row_checksum) | 用于查询行数据的 Checksum 值。该函数只能用于 FastPlan 流程的 `SELECT` 语句，即你可通过类似 `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id = ?` 或 `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id IN (?, ?, ...)` 的语句进行查询。参见[数据正确性校验](/ticdc/ticdc-integrity-check.md)。 |
+| [`TIDB_SHARD()`](#tidb_shard) | 用于创建一个 SHARD INDEX 来打散热点索引。SHARD INDEX 是一种以 `TIDB_SHARD` 函数为前缀的表达式索引。 |
+| [`TIDB_VERSION()`](#tidb_version) | 用于获取当前连接的 TiDB 服务器的版本以及构建信息。 |
+| [`VITESS_HASH()`](#vitess_hash) | 返回与 Vitess 的 `HASH` 函数兼容的字符串哈希值，有助于从 Vitess 迁移至 TiDB。 |
```
Comment (lilin90) [inline functions-and-operators/tidb-functions.md:26]:
```suggestion
| [`VITESS_HASH()`](#vitess_hash) | 返回与 Vitess 的 `HASH` 函数兼容的数值的哈希值，有助于从 Vitess 迁移至 TiDB。 |
```

**Review by easonn7 — APPROVED** [review]

File: functions-and-operators/tidb-functions.md [hunk 2]
```diff
@@ -9,34 +9,146 @@ summary: 学习使用 TiDB 特有的函数。
...
+1 row in set (0.00 sec)
+```
+
+## TIDB_BOUNDED_STALENESS
+
+`TIDB_BOUNDED_STALENESS()` 函数用作 [`AS OF TIMESTAMP`](/as-of-timestamp.md) 语法的一部分。
+
+## TIDB_CURRENT_TSO
+
+`TIDB_CURRENT_TSO()` 函数返回当前事务的 TSO，类似于 [`tidb_current_ts`](/system-variables.md#tidb_current_ts) 变量。
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:82]:
```suggestion
`TIDB_CURRENT_TSO()` 函数返回当前事务的 [TSO](/tso.md)，类似于 [`tidb_current_ts`](/system-variables.md#tidb_current_ts) 变量。
```

File: functions-and-operators/tidb-functions.md [hunk 3]
```diff
@@ -9,34 +9,146 @@ summary: 学习使用 TiDB 特有的函数。
...
-### TIDB_DECODE_KEY
+1 row in set (0.00 sec)
+```
 
-`TIDB_DECODE_KEY` 函数用于将 TiDB 编码的键输入解码为包含 `_tidb_rowid` 和 `table_id` 的 JSON 结构。你可以在一些系统表和日志输出中找到 TiDB 的编码键。
+## TIDB_DECODE_KEY
+
+`TIDB_DECODE_KEY()` 函数用于将 TiDB 编码的键输入解码为包含 `_tidb_rowid` 和 `table_id` 的 JSON 结构。在一些系统表和日志输出中有 TiDB 编码的键。
 
 以下示例中，表 `t1` 有一个隐藏的 `rowid`，该 `rowid` 由 TiDB 生成。语句中使用了 `TIDB_DECODE_KEY` 函数。结果显示，隐藏的 `rowid` 被解码后并输出，这是典型的非聚簇主键结果。
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:145]:
```suggestion
以下示例中，表 `t1` 有一个隐藏的 `rowid`，该 `rowid` 由 TiDB 生成。语句中使用了 `TIDB_DECODE_KEY()` 函数。结果显示，隐藏的 `rowid` 被解码后并输出，这是典型的非聚簇主键结果。
```

File: functions-and-operators/tidb-functions.md [hunk 4]
```diff
@@ -140,70 +252,22 @@ ORDER BY
...
-1 row in set (0.00 sec)
-```
-
-### TIDB_DECODE_SQL_DIGESTS
+## TIDB_DECODE_SQL_DIGESTS
 
 `TIDB_DECODE_SQL_DIGESTS()` 函数用于在集群中查询一组 SQL Digest 所对应的 SQL 语句的归一化形式（即去除格式和参数后的形式）。函数接受 1 个或 2 个参数：
 
 * `digests`：字符串类型，该参数应符合 JSON 字符串数组的格式，数组中的每个字符串应为一个 SQL Digest。
 * `stmtTruncateLength`：可选参数，整数类型，用来限制返回结果中每条 SQL 语句的长度，超过指定的长度会被截断。0 表示不限制长度。
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:268]:
```suggestion
* `stmtTruncateLength`：可选参数，整数类型，用来限制返回结果中每条 SQL 语句的长度，超过指定的长度会被截断。`0` 表示不限制长度。
```

File: functions-and-operators/tidb-functions.md [hunk 5]
```diff
@@ -140,70 +252,22 @@ ORDER BY
...
 
-返回一个字符串，符合 JSON 字符串数组的格式，数组中的第 *i* 项为参数 `digests` 中的第 *i* 个元素所对应的语句。如果参数 `digests` 中的某一项不是一个有效的 SQL Digest 或系统无法查询到其对应的 SQL 语句，则返回结果中对应项为 `null`。如果指定了截断长度（`stmtTruncateLength > 0`），则返回结果中每条超过该长度的语句，保留前 `stmtTruncateLength` 个字符，并在尾部增加 `"..."` 后缀表示发生了截断。如果参数 `digests` 为 `NULL`，则函数的返回值为 `NULL`。
+该函数返回一个字符串，符合 JSON 字符串数组的格式，数组中的第 *i* 项为参数 `digests` 中的第 *i* 个元素所对应的语句。如果参数 `digests` 中的某一项不是一个有效的 SQL Digest 或系统无法查询到其对应的 SQL 语句，则返回结果中对应项为 `null`。如果指定了截断长度（`stmtTruncateLength > 0`），则返回结果中每条超过该长度的语句，保留前 `stmtTruncateLength` 个字符，并在尾部增加 `"..."` 后缀表示发生了截断。如果参数 `digests` 为 `NULL`，则函数的返回值为 `NULL`。
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:270]:
```suggestion
该函数返回一个字符串，符合 JSON 字符串数组的格式，数组中的第 *i* 项为参数 `digests` 中的第 *i* 个元素所对应的语句。如果参数 `digests` 中的某一项不是一个有效的 SQL Digest 或系统无法查询到其对应的 SQL 语句，则返回结果中对应项为 `null`。如果指定了截断长度 (`stmtTruncateLength > 0`)，则返回结果中每条超过该长度的语句，保留前 `stmtTruncateLength` 个字符，并在尾部增加 `"..."` 后缀表示发生了截断。如果参数 `digests` 为 `NULL`，则函数的返回值为 `NULL`。
```

File: functions-and-operators/tidb-functions.md [hunk 6]
```diff
@@ -249,62 +313,114 @@ select tidb_decode_sql_digests(@digests, 10);
 - [`Statement Summary Tables`](/statement-summary-tables.md)
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:313]:
```suggestion
- [Statement Summary Tables](/statement-summary-tables.md)
```

File: functions-and-operators/tidb-functions.md [hunk 7]
```diff
@@ -249,62 +313,114 @@ select tidb_decode_sql_digests(@digests, 10);
...
+1 row in set (0.0012 sec)
+```
+
+以上示例使用 `TIDB_PARSE_TSO()` 函数从 `tidb_current_ts` 会话变量提供的可用时间戳编号中提取物理时间戳。因为每个事务都会分配到时间戳，所以此函数在事务中运行。
 
-### TIDB_ROW_CHECKSUM
+## TIDB_PARSE_TSO_LOGICAL
 
-`TIDB_ROW_CHECKSUM` 函数用于查询行数据的 Checksum 值。该函数只能用于 FastPlan 流程的 `SELECT` 语句，即你可通过形如 `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id = ?` 或 `SELECT TIDB_ROW_CHECKSUM() FROM t WHERE id IN (?, ?, ...)` 的语句进行查询。
+`TIDB_PARSE_TSO_LOGICAL(tso)` 函数返回从 TiDB TSO 时间戳中提取逻辑时间戳。
```
Comment (Oreoxmt) [inline functions-and-operators/tidb-functions.md:393]:
```suggestion
`TIDB_PARSE_TSO_LOGICAL(tso)` 函数返回从 TiDB [TSO](/tso.md) 时间戳中提取逻辑时间戳。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**